release/release.go

```diff
@@ -113,6 +116,25 @@ func isWSL() bool {
 	return fileExists("/proc/sys/fs/binfmt_misc/WSLInterop")
 }
 
+var ioutilReadFile = ioutil.ReadFile
+
+// WSL 1 /proc/version: Linux version [kernel]-Microsoft (Microsoft@Microsoft.com) ([gcc version] ) #[build]-Microsoft [timestamp]
+// WS2 2 /proc/version: Linux version [kernel]-microsoft-standard-WSL2 (oe-user@oe-host) ([gcc version], [ld version]) #1 [timestamp]
+//                                             ^ We use Microsoft (upper case M) to distinguish them
+//
+// Note that WS2's kernel can be customized by the user, so there is no guarantee on the
+// contents of that string. Hence why we search in WSL1's kernel name instead of WSL2's
```

Comment:
This seems rather fragile, if the user renames the kernel to something with `Microsoft` in it then we would assume it's v1.
If really there isn't a better way, I would suggest at least to use a regexp like (untested):
```
-Microsoft .* #[0-9.]*-Microsoft
```
(assuming that the build number only contains digits and dots)

Reply:
Here are a couple examples of WSL1 Kernels:
```
Linux version 3.4.0-Microsoft (Microsoft@Microsoft.com) (gcc version 4.7 (GCC) ) #1 SMP PREEMPT Wed Dec 31 14:42:53 PST 2014
Linux version 4.4.0-22000-Microsoft (Microsoft@Microsoft.com) (gcc version 5.4.0 (GCC) ) #653-Microsoft Wed Apr 27 16:06:00 PST 2022
```

I'll go with regex:
```
Linux version [0-9.-]+Microsoft \(Microsoft@Microsoft.com\)
```

Reply:
The most reliable way of distinguishing WSL 1 from WSL 2 is to look at the type of the root filesystem. Under WSL1, this will always be `wslfs` or `lxfs` (under early WSL1 releases), which filesystems only exist under WSL 1. If it's `ext4` or anything else, then it's WSL 2.

(Presuming that you have already established that you are running under WSL. For this, a convenient means is checking for the `/run/WSL` directory, which exists under both varieties of WSL regardless of custom kernel, whether Windows interoperability is enabled in wsl.conf - this will break the current WSL detection mechanism in this PR - etc.)

Reply:
I have not considered the option of looking at the root filesystem, I'll get back to you on that but it does indeed sound more robust.

Regarding the second point, even with interop disabled the current mechanism works, as disabling interop does not remove file `/proc/sys/fs/binfmt_misc/WSLInterop`. It also exists both for WSL1 and WSL2.

Reply:
I tested it by enabling it and disabling it as well, I don't have any stronger evidence. If it does disappear in your system, the that's enough reason to switch to your method. Please do report on your findings.

Reply:
Okay, I just tested this again, with inconsistent results. I started out grabbing a spare machine and making a couple of clean Ubuntu installs to minimize potential interference. What I found was that under WSL 1, disabling interop made it disappear, but under WSL 2, it didn't.

On my main machine, it appears to disappear under WSL 2 but only if native _systemd_ support is enabled.

This I believe is an effect from enabling the native _systemd_ support. While this may change as the native support becomes less experimental, the current state of affairs appears to be that since _systemd_ mounts API filesystems, including `/proc/sys/fs/binfmt_misc`, on startup, this replaces the WSL binfmt_misc, and loses _WSLInterop_ in the process.

(This was an issue in previous _systemd_ hacks for WSL as well. Both my _systemd_-enabler for earlier WSL releases, _genie_, and my current issue-fixer for _systemd_ on WSL, _bottle-imp_, get around this by adding an explicit _WSLInterop.conf_ file to _/lib/binfmt.d_ to restore it, but this still leaves a window during instance startup in which it doesn't exist.)

...it might be worth moving to `/run/WSL` just to avoid having to deal with all this? 😬

Reply:
> ...it might be worth moving to /run/WSL just to avoid having to deal with all this? 😬

Definitely, I'll start a separate PR to fix that.